### ISSUE TITLE
[Bug]: could not load library timescaledb-2.7.0.so undefined symbol: pull_varnos_new

### ISSUE THREAD

**Issue by fetchezar** [timescale/timescaledb#4409]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Platform/OS, Other

### What happened?

Hi! We're in the process of updating our database to TimescaleDB 2.x.x

We have two databases, both originally with TimescaleDB 1.7.x, and PosgreSQL 12

We wanted to upgrade both TimescaleDB to 2.x and the PostgreSQL version to 14. What's recommended in the site is to upgrade PostgreSQL first however this was not possible in our case since TimescaleDB 1.7.x doesn't supports PostgreSQL 14.

So we proceeded to do the reverse steps, upgrade first the TimescaleDB extension to whatever was the latest at the moment, and then to PostgreSQL 14.

On the first database this went great, the latest tsdb version was 2.6.1 at the time. I'm not exactly sure what the exact version of postgres we had, but I believe it was this one:

`postgresql-12 12.10-1.pgdg100+1`

So we ended up with this database succesfully upgraded to PostgreSQL 14 and TimescaleDB 2.6.1

We're in the process of trying to update our second database, but we encountered a problem.

First issue was that 2.7.0 got released, so we updated our packages and that's the one we got. We decided to try it anyway and gave it a go.

In the process of updating the extension, after updating the packages, and restarting the database for good measure, we tried to do the following:

`ALTER EXTENSION timescaledb UPDATE;` 

Which results in the following error on our part

> ERROR: could not load library "/usr/lib/postgresql/12/lib/timescaledb-2.7.0.so": /usr/lib/postgresql/12/lib/timescaledb-2.7.0.so: undefined symbol: pull_varnos_new

We thought this just may be a new problem, 2.7.0 was pretty new after all, so we forced apt to install 2.6.1 instead, and tried again. Also specifying the version to 2.6.1, but we get the same error log. There is an undefined symbol in the .so, pull_varnos_new.

The only difference we could find with the first database, where the upgrade went well, was the PostgreSQL version, which is the following one in this database that is failing:

`postgresql-12 12.11-1.pgdg100+1`

So there is a slight difference, this one is a bit newer.

Our debian version may be a bit old my guess is that we'd have a missing symbol in some other system lib, or even maybe in PostgreSQL itself, instead of missing an exported function in the TimescaleDB .so itself. But I dont really know so I was looking for some feedback on this.

Thank you for your time.

### TimescaleDB version affected

2.7.0

### PostgreSQL version used

12.11

### What operating system did you use?

Debian 10 x64 Buster

### What installation method did you use?

Deb/Apt

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
ERROR: could not load library "/usr/lib/postgresql/12/lib/timescaledb-2.7.0.so": /usr/lib/postgresql/12/lib/timescaledb-2.7.0.so: undefined symbol: pull_varnos_new
```

### How can we reproduce the bug?

```bash
Install `postgresql-12 12.11-1.pgdg100+1`
Install TimescaleDB 1.7.2
Install TimescaleDB 2.7.0 or 2.6.1
Try to update the extension
```

**Comment by fetchezar** [comment]
We're using these repos, if it helps:

> deb http://apt.postgresql.org/pub/repos/apt/ buster-pgdg main
deb https://packagecloud.io/timescale/timescaledb/debian/ buster main

**Comment by fetchezar** [comment]
Reading the code a bit I saw this
https://github.com/timescale/timescaledb/blob/35ea80ffdffcdcf3c81af7881c509f33bbeb9924/src/compat/compat.h#L105

That made me track down the specific version we're running and I think I found the problem. Our online PostgreSQL instance is actually 12.4. Apparently the PostgreSQL package got updated, but the instance was never restarted with the new binaries.

I believe the timescaledb package never complained about this because the apt dependency was met, the installed PostgreSQL version was in fact >= 12.8 (which is the minimum the timescaledb package is declaring at the time of writing).

I'm not exactly sure if it's a feasible suggestion but maybe TimescaleDB could check this at runtime? If the specific online version is supported I mean.

In any case it looks the problem was on our end, we'll be testing again the upgrade at an appropriate time and will report back if we continue to have these issues.
